### DISCUSSION TITLE
4.1] identification with email

### DISCUSSION THREAD

**Discussion by HDInfautre** [joomla/joomla-cms#36119] [2. Cold Features]
Hi
Like many users, I put a different nickname for each site.
It's hard to find it when you don't log in often.
Many sites accept email authentication.
There was a plugin in joomla 3 that a priori will not be ported to joomla 4
Request:
Allow identification with email. This probably involves better control of the creation of a new member with a nickname that is not an email (and perhaps rejected some term (admin ....)
[I'm not a developer and I don't have a technical proposal to make. If I am not in the right place, please move or tell me how to do it].

**Comment by lancedouglas1** [comment]
The way I dealt with this in J3 is to create my own plugin that overrides the user registration and update functions.  I basically hide the "username" field from the forms and only use email.  I force the username that is stored in the database to be the email address.

In J4 I am currently finishing migration testing and will be using a 3rd party OAuth registration and login solution so I'll be skipping the entire joomla core user creation and registration functions.

**Reply by HDInfautre** [reply]
Hi
Thank you for this information. I don't understand why you had to rewrite the joomla? I find it dangerous to use only one email. 
I had an email linked to a domain name that I deleted. The problem was that I had problems with sites that only allowed identification by this email :-(

I don't understand why, email recognition is not part of core joomla. It is a common practice of members who forget their login. I think it's important to give them an alternative.
